File: Day #1820/solution.js
// String rotation check: B is a rotation of A iff |A|==|B| and B is a substring of A+A.
// Time: O(n) (includes). Space: O(n).
function isRotation(a, b) {
  return a.length === b.length && (a + a).includes(b);
}

console.log(isRotation("abcde", "cdeab")); // true
console.log(isRotation("abc", "acb"));     // false

File: Day #1820/solution.py
# Day 1820: String rotation check: B is a rotation of A iff |A|==|B| and B is a substring of A+A.
# Time: O(n) (in operator). Space: O(n).


def is_rotation(a, b):
    return len(a) == len(b) and b in (a + a)


if __name__ == "__main__":
    print(str(is_rotation("abcde", "cdeab")).lower())  # true
    print(str(is_rotation("abc", "acb")).lower())      # false
